src/Sarif.Viewer.VisualStudio/Options/SarifViewerOptions.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+using Microsoft.VisualStudio.Shell;
+
+namespace Microsoft.Sarif.Viewer.Options
+{
+    internal class SarifViewerOption
+    {
+        private readonly bool shouldMonitorSarifFolderDefaultValue = true;
+
+        private readonly AsyncPackage package;
+
+        private readonly SarifViewerOptionPage optionPage;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SarifViewerOption"/> class.
+        /// Get visual studio option values.
+        /// </summary>
+        /// <param name="package">Owner package, not null.</param>
+        private SarifViewerOption(AsyncPackage package)
+        {
+            this.package = package ?? throw new ArgumentNullException(nameof(package));
+            this.optionPage = (SarifViewerOptionPage)this.package.GetDialogPage(typeof(SarifViewerOptionPage));
+        }
+
+        private SarifViewerOption()
+        {
+        }
+
+        public bool ShouldMonitorSarifFolder
+        {
+            get
+            {
+                try
+                {
+                    return this.optionPage != null ? this.optionPage.MonitorSarifFolder : this.shouldMonitorSarifFolderDefaultValue;
```

Comment:
>this.optionPage [](start = 27, length = 15)

I think you could change to:
return this.optionPage?.MonitorSarifFolder ?? this.shouldMonitorSarifFolderDefaultValue

Reply:
Also, i think you don't need the try/catch, unless you are expecting something else

---
In reply to: [597772222](https://github.com/microsoft/sarif-visualstudio-extension/pull/382#discussion_r597772222) [](ancestors = 597772222)

Reply:
fixed

---
In reply to: [597772640](https://github.com/microsoft/sarif-visualstudio-extension/pull/382#discussion_r597772640) [](ancestors = 597772640,597772222)